src/geometry_aux.cpp

```diff
@@ -40,6 +42,71 @@ void update_universe_cell_count(int32_t a, int32_t b)
   }
 }
 
+double calculate_optical_thickness(Position start_sampled_position,
+  Position end_sampled_position, uint64_t* seed)
+{
+  Position direction = (end_sampled_position - start_sampled_position) /
+                       (end_sampled_position - start_sampled_position).norm();
+
+  Particle p;
+  SourceSite site;
+  site.E = 1.0;
+  site.particle = ParticleType::neutron;
+  site.r = start_sampled_position;
+  site.u = direction;
+  p.from_source(&site);
+
+  double optical_thickness = 0.0;
+
+  while (true) {
+    if (!exhaustive_find_cell(p))
+      break;
+
+    BoundaryInfo boundary = distance_to_boundary(p);
+    double distance_to_boundary = boundary.distance;
```

Comment:
Don't need to store this `Boundary` object
```suggestion
    double distance_to_boundary = distance_to_boundary(p).distance;
```